apps/core/scenes/search/Search.js

```diff
@@ -55,42 +78,114 @@ export default class Search extends React.Component<Props, State> {
     console.warn('TODO'); // eslint-disable-line no-console
   };
 
+  handleTripTypeSelect = (type: string) => {
+    this.setState({ tripType: type, modalType: MODAL_TYPE.HIDDEN });
+  };
+
+  handleTripTypePress = () => {
+    this.setState({ modalType: MODAL_TYPE.TRIP_TYPE });
+  };
+
+  handlePassengersPress = () => {
+    this.setState({ modalType: MODAL_TYPE.PASSENGERS });
+  };
+
+  handleModalClose = () => {
+    this.setState({ modalType: MODAL_TYPE.HIDDEN });
+  };
+
+  getModalContent = () => {
```

Comment:
what about `renderModalContent` ?

Reply:
Thanks, make sense, `render` is definitely more consistent.
Updated